lib/phoenix/live_dashboard/helpers/supervision_tree_helpers/supervision_tree.ex

```diff
@@ -0,0 +1,114 @@
+defmodule Phoenix.LiveDashboard.SupervisionTree do
+  # Construct supervision tree given an application master
+  @moduledoc false
+  alias Phoenix.LiveDashboard.{SystemInfo}
+
+  @node_width 100
+  @node_x_separation 50
+  @total_x_distance @node_width + @node_x_separation
+
+  def construct_tree(master) do
```

Comment:
I am thinking it may be best to convert this into two distinct steps:

1. First we get the whole tree from SystemInfo. The tree will look like this:

        @type tree :: {{pid, name}, children :: [tree()]}

2. Then, when we first pass this tree to `ReingoldTilford`, we fully convert it to ReingoldTilford's internal representation. This way we can keep those two fully decoupled from each other.

Moving into this direction will also make it easier to unit test stuff, as we can unit test `ReingoldTilford` without worrying about the supervision tree and we can unit test supervision tree without worrying about `ReingoldTilford`.

WDYT?

Reply:
Yes, I think it's the best way